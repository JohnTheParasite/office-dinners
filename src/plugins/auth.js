import AuthService from "@/services/authService";

export const Auth = {
  install(Vue) {
    Vue.prototype.$authService = AuthService;
  },
};