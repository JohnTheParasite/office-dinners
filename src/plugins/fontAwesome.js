import { library } from "@fortawesome/fontawesome-svg-core"
import {
  faArrowRightToBracket,
  faArrowUpRightFromSquare,
  faBan,
  faBurger,
  faChartLine,
  faCheck,
  faChevronDown,
  faChevronLeft,
  faChevronRight,
  faEye,
  faEyeSlash,
  faHandHoldingDollar,
  faHeart,
  faPencil,
  faPercent,
  faTrashCan,
  faUser,
  faUsers,
  faUtensils,
  faXmark
} from "@fortawesome/free-solid-svg-icons"
import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome"
import Vue from "vue"

library.add(faChevronLeft)
library.add(faChevronRight)
library.add(faChevronDown)
library.add(faArrowUpRightFromSquare)
library.add(faCheck)
library.add(faPencil)
library.add(faTrashCan)
library.add(faBan)
library.add(faPercent)
library.add(faChartLine)
library.add(faUsers)
library.add(faBurger)
library.add(faHandHoldingDollar)
library.add(faUtensils)
library.add(faHeart)
library.add(faUser)
library.add(faArrowRightToBracket)
library.add(faXmark)
library.add(faEye)
library.add(faEyeSlash)
Vue.component("font-awesome-icon", FontAwesomeIcon)
