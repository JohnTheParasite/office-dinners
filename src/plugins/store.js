import store from "@/store/store"

export const Store = {
  install(Vue) {
    Vue.prototype.$store = store
  }
}
