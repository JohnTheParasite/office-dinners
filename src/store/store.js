import Vue from "vue"
import Vuex from "vuex"
import global from "./modules/global"
import toasts from "./modules/toasts"

Vue.use(Vuex)

export default new Vuex.Store({
  strict: true,
  state: {},
  mutations: {},
  actions: {},
  modules: {
    global,
    toasts
  }
})
