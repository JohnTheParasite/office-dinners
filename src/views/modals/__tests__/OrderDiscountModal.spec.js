import { createLocalVue, mount, shallowMount } from "@vue/test-utils"
import orderDiscountModal from "@/views/modals/OrderDiscountModal"
import i18n from "@/i18n"
import store from "@/store/store"
import { BootstrapVue } from "bootstrap-vue"
import "@/plugins/fontAwesome"

describe("orderDiscountModal", () => {
  const localVue = new createLocalVue()
  localVue.use(i18n)
  localVue.use(store)
  localVue.use(BootstrapVue)

  it("should render properly", () => {
    let orderDiscount = shallowMount(orderDiscountModal, { localVue, i18n, store })
    expect(orderDiscount).toMatchSnapshot()
  })

  it("should test onChange and show method", async () => {
    let orderDiscount = mount(orderDiscountModal, { localVue, i18n, store })
    expect(orderDiscount.vm.discount).toBe(0)
    orderDiscount.vm.onChange("discount", 5)
    expect(orderDiscount.vm.discount).toBe(5)
    expect(orderDiscount.vm.$children[0].visible).toBeFalsy()
    orderDiscount.vm.show()
    await orderDiscount.vm.$nextTick()
    expect(orderDiscount.vm.discount).toBe(0)
    expect(orderDiscount.vm.$children[0].isVisible).toBeTruthy()
    orderDiscount.vm.hide()
    await orderDiscount.vm.$nextTick()
    expect(orderDiscount.vm.$children[0].isVisible).toBeFalsy()
  })
})
