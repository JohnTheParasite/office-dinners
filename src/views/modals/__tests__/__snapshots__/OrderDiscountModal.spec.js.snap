// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`orderDiscountModal should render properly 1`] = `
<b-modal-stub id="b6" canceltitle="Cancel" cancelvariant="secondary" centered="true" headerclass="warning" headerclosecontent="&amp;times;" headercloselabel="Close" nocloseonbackdrop="true" oktitle="OK" okvariant="primary" size="md" titletag="h5">
  <div class="content">
    <div class="discount-area">
      <div class="discount-input">
        <text-input-stub label="" description="" name="" type="number" placeholder="" warningmessage="errors.fieldIsRequired" icon="" oniconclick="[Function]"></text-input-stub>
      </div>
      <toggle-stub label="interface.percent" description="" name=""></toggle-stub>
    </div>
  </div>
</b-modal-stub>
`;
